legacy-device/src/serial.rs

```diff
@@ -190,8 +191,17 @@ impl Serial {
     }
 }
 
-impl BusDevice for Serial {
-    fn read(&mut self, offset: u64, data: &mut [u8]) {
+impl Device for Serial {
+    fn name(&self) -> String {
+        "Serial".to_string().clone()
+    }
+
+    fn set_resources(&mut self, _res: &[IoResource], _irq: Option<IrqResource>) {
```

Comment:
Ummm, understood. I remembered if we have specify an address to DeviceManager::register_device, it won't re-allocate for us. So the address won't change.  So do we still need get back address?

Reply:
We will always get the address back. Now do we need to compare? Actually not, as the registration would have failed otherwise. We can leave that one empty.

Reply:
Agreed.